src/main/python/findDist/findDist.py

```diff
@@ -1,71 +1,45 @@
-from shapely.geometry import Polygon,shape, MultiPolygon,LineString,MultiLineString,Point
-from shapely.validation import explain_validity
+from shapely.geometry import shape,Point
 
 class Precinct:
 	def __init__(self,precinctId,center):
 		self.precinctId = precinctId
 		self.center = center
 
-
 class District:
 	def __init__(self,districtId,polygon):
 		self.districtId = districtId
 		self.polygon = polygon	
 
-def findCoordByID(idd,searchRange):
-	for item in searchRange:
-		if item['properties']['GEOID10'] == idd:
-			return item['geometry']
+inputFile = sys.argv[1]
+outputFile = sys.argv[2]
+state = sys.argv[3].upper()+'-'
+numOfDistricts = int(sys.argv[4])
 
 precincts = []
-with open(sys.argv[1],'r') as f:
-	data = json.load(f)['features']
-	for item in data:
-		precinctId = item['properties']['GEOID10']
-		precinctCenter = [float(item['properties']['INTPTLON10']),float(item['properties']['INTPTLAT10'])]
-		precincts.append(Precinct(precinctId,Point(precinctCenter)))
-f.close()
-
+with open(inputFile,'r') as fPrecinct:
+	data = json.load(fPrecinct)['features']
+	precincts = list(map(lambda item : Precinct(item['properties']['GEOID10'],
+				Point([float(item['properties']['INTPTLON10']),float(item['properties']['INTPTLAT10'])])),data))
+fPrecinct.close()
 
 districts = []
-for i in range(1,int(sys.argv[4])+1):
-	with open('./'+sys.argv[3]+'-'+str(i)+'/shape.geojson') as f:
-		data=json.load(f)
+for i in range(1,numOfDistricts+1):
+	with open(state+str(i)+'/shape.geojson') as fDistrict:
+		data=json.load(fDistrict)
 		districts.append(District(data['properties']['Code'],shape(data['geometry'])))
-f.close()
-
-f = open(sys.argv[2],"w+")
-f.write("[\n")
-outOfBoundary=[]
-for precinct in precincts:
-	for district in districts:
-		if district.polygon.contains(precinct.center):
-			if precinct == precincts[-1]:
-				f.write('{"id":"'+str(precinct.precinctId)+'","district":"'+str(district.districtId)+'"}')
-			else:
-				f.write('{"id":"'+str(precinct.precinctId)+'","district":"'+str(district.districtId)+'"},\n')
-			break
-		elif not district.polygon.contains(precinct.center) and district==districts[-1]:
-			outOfBoundary.append(precinct)
+fDistrict.close()
 
-
-if(outOfBoundary == []):
-	f.write(']')
-else:
-	f.write(',\n')
-	for precinct in outOfBoundary:
-		disId = None
-		minn = sys.maxsize
+jsonStrings = []
+with open(outputFile,"w") as fOutput:
+	for precinct in precincts:
 		for district in districts:
-			distance = precinct.center.distance(district.polygon)
-			if distance<minn:
-				disId = district.districtId
-				minn = distance
-		if precinct == outOfBoundary[-1]:
-			f.write('{"id":"'+str(precinct.precinctId)+'","district":"'+disId+'"}]\n')
-		else:
-			f.write('{"id":"'+str(precinct.precinctId)+'","district":"'+disId+'"},\n')
-
+			if district.polygon.contains(precinct.center): #find a district
+				jsonStrings.append(json.dumps({"id":precinct.precinctId,"district":district.districtId}))
+				break
+	fOutput.write('[')

```

Comment:
Same problem as on the other file.
We don't want to do low level string manipulation like this.